shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/keygen/CosIdKeyGenerateAlgorithm.java

```diff
@@ -20,57 +20,47 @@
 import me.ahoo.cosid.CosId;
 import me.ahoo.cosid.provider.IdGeneratorProvider;
 import me.ahoo.cosid.provider.LazyIdGenerator;
-import org.apache.shardingsphere.infra.config.algorithm.ShardingSphereInstanceRequiredAlgorithm;
-import org.apache.shardingsphere.infra.instance.InstanceContext;
 import org.apache.shardingsphere.sharding.algorithm.sharding.cosid.CosIdAlgorithm;
 import org.apache.shardingsphere.sharding.spi.KeyGenerateAlgorithm;
 
-import java.util.Optional;
 import java.util.Properties;
 
 /**
  * CosId key generate algorithm.
  */
-public final class CosIdKeyGenerateAlgorithm implements KeyGenerateAlgorithm, ShardingSphereInstanceRequiredAlgorithm {
```

Comment:
@Ahoo-Wang Why remove ShardingSphereInstanceRequiredAlgorithm interface?

Reply:
Because the `SegmentId` algorithm and `SegmentChainId` algorithm of `CosId` do not need to depend on `WorkerId`, so I added `CosIdSnowflakeKeyGenerateAlgorithm`.

Reply:
> 

Thank you, I got it.